server/datastore/mysql/software.go

```diff
@@ -547,7 +547,14 @@ func (ds *Datastore) getIncomingSoftwareChecksumsToExistingTitles(
 			argsWithoutBundleIdentifier = append(argsWithoutBundleIdentifier, sw.Name, sw.Source, sw.Browser)
 		}
 		// Map software title identifier to software checksums so that we can map checksums to actual titles later.
-		uniqueTitleStrToChecksum[UniqueSoftwareTitleStr(sw.Name, sw.Source, sw.Browser)] = checksum
+		uniqueTitleStrToChecksum[UniqueSoftwareTitleStr(
+			func() string {
```

Comment:
So, `UniqueSoftwareTitleStr` is used in a handful of other places, and in those places it uses software name rather than bundle ID. Thinking that those other places should take the new uniqueness constraint into account to display/deduplicate things properly.

Probably need to check references to `fleet.SoftwareFieldSeparator` because those may be hand-building the same uniqueness constraint so we'd need to fix those too.

Reply:
RE: `fleet.SoftwareFieldSeparator`
Besides tests here are the only locations:
https://github.com/fleetdm/fleet/blob/main/server/datastore/mysql/software.go#L150
https://github.com/fleetdm/fleet/blob/main/server/datastore/mysql/software.go#L161
Both of those seem unrelated.

https://github.com/fleetdm/fleet/blob/main/server/datastore/mysql/software.go#L618 The `UniqueSoftwareTitleStr` instance

Reply:
Yep, the L150/L161 usages are indeed unrelated. Thanks for checking!